cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -474,9 +480,10 @@ constexpr uint32_t encoding_to_mask(Encoding encoding)
  * Used to control which encode kernels to run.
  */
 enum class encode_kernel_mask {
-  PLAIN        = (1 << 0),  // Run plain encoding kernel
-  DICTIONARY   = (1 << 1),  // Run dictionary encoding kernel
-  DELTA_BINARY = (1 << 2)   // Run DELTA_BINARY_PACKED encoding kernel
+  PLAIN           = (1 << 0),  // Run plain encoding kernel
+  DICTIONARY      = (1 << 1),  // Run dictionary encoding kernel
+  DELTA_BINARY    = (1 << 2),  // Run DELTA_BINARY_PACKED encoding kernel
+  DELTA_LENGTH_BA = (1 << 3),  // Run DELTA_LENGTH_BYTE_ARRAY encoding kernel
```

Comment:
It will make room. Delta byte array is Klingon!

Reply:
This was a mildly serious question since as the enums are ATM we'll eventually end up with different order in `decode_kernel_mask` and `encode_kernel_mask`.
But I realized that these are internal and we can modify the assigned values at any point.

Reply:
I'm not sure I follow...DELTA_BYTE_ARRAY will be bit 4. The only other possible encoding right now would be BYTE_STREAM_SPLIT which is for floats (but there's a proposal to extend it to FIXED_LEN_BYTE_ARRAY), and that would probably be implemented in the PLAIN encoder anyway. If the mask is 32 bits, that leaves plenty of room for other flags, like NO_NULLS, NO_REPETITION, etc which would be more in line with @nvdbaranec's original vision for the smaller special purpose kernels.